### PR TITLE
fix(packagesigning): filter out nodes using containers, thanks to installation method

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Prevent sending package signing telemetry in nodes installed in a container-like environment

### Motivation
Prevent sending unused information

### Additional Notes
List of installation methods can be found on [metabase](https://metabase-analytics.us1.prod.dog/question/23861-install-method-usage?Install_method=)

### Describe how to test/QA your changes
Automated tests should check non-regression.
Start a docker-agent and execute
```
docker --context pulumi-10.1.x.x exec -it <container_id> agent diagnose show-metadata package-signing
```
You should get an error `Error: Could not fetch metadata v5 payload: {"error":"inventory metadata is disabled"}`
Same error when `enable_metadata_package_signing` = False or install_tool tool is changed on a linux vm

### CODE REVIEW

File: comp/metadata/packagesigning/packagesigningimpl/packagesigning.go [hunk 1]
```diff
@@ -128,11 +130,29 @@ func newPackageSigningProvider(deps dependencies) provides {
...
+		logger.Debugf("Failed to get install_info file information: %v", err)
 		return false
 	}
+	isInConfigurationFile := conf.GetBool("enable_signing_metadata_collection")
+	isEnabled := isInConfigurationFile && runtime.GOOS == "linux" && isAllowedInstallationTool(installTool.Tool)
+	if !isEnabled {
+		logger.Debugf("Package signing metadata collection disabled: config %t, OS %s, install tool %s", isInConfigurationFile, runtime.GOOS, installTool.Tool)
+		logger.Debug("Package signing telemetry must be enabled in datadog.yaml, and running on a non-containerized Linux system to collect data")
+	}
 	logger.Debug("Signing metadata collection enabled")
```
Comment (chouquette) [inline comp/metadata/packagesigning/packagesigningimpl/packagesigning.go:144]: nit: shoudln't this be in an `else` block?

**Review by chouquette — APPROVED** [review]
I don't see anything wrong, but I might lack some context. I'd encourage you to get another pair of eyes on the PR before merging :)

File: comp/metadata/packagesigning/packagesigningimpl/packagesigning.go (hunk 1 above)
Comment (chouetz) [reply comp/metadata/packagesigning/packagesigningimpl/packagesigning.go:144]: correct, thanks

**Pull request merged**